sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_encode.py

```diff
@@ -303,6 +304,37 @@ def encode_double(output: bytearray, value: float, with_constructor: bool = True
     output.extend(struct.pack(">d", value))
 
 
+def encode_decimal128(output: bytearray, value: Decimal, with_constructor: bool = True, **kwargs: Any) -> None:
```

Comment:
should we support other decimal types now too? If we do 128 it feels like we should support all AMQP does

Reply:
I think we can prob wait on this until there's a need? Since we're not planning on making pyamqp external yet and the services don't support other ones. Could we maybe just add a TODO comment as a placeholder for those so we know that needs to be done before pyamqp external GA?